recipe/build_clangdev.sh

```diff
@@ -1,3 +1,10 @@
+
+sed -i.bak -e 's@addSystemInclude(DriverArgs, CC1Args, SysRoot + "/usr/local/include");@addSystemInclude(DriverArgs, CC1Args, "'"${PREFIX}/x86_64-conda_cos6-linux-gnu/sysroot/usr/include"'");@g' \
+	    lib/Driver/ToolChains/Linux.cpp && rm $_.bak
+
+sed -i.bak -e 's@AddPath("/usr/local/include", System, false);@AddPath("'"${PREFIX}/x86_64-conda_cos6-linux-gnu/sysroot/usr/include"'", System, false);@g' \
+	    lib/Frontend/InitHeaderSearch.cpp && rm $_.bak
```

Comment:
Interesting! I am curious what this is about exactly. This might be the root to another issue we have in cling.

Reply:
I might have slightly misunderstood how I managed to fix this but when bootstrapping `rootcling` the first stage is useless as it tries to find that standard headers from a weird mixture of the `/usr` and `/prefix`. This causes `#include_next` to fail to find several headers (such as `features.h`).

Reply:
Use `$HOST` instead of `x86_64-conda_cos6-linux-gnu`

Reply:
Can you add the directory `$PREFIX/$HOST/usr/include` as additionally to `/usr/local/include`?

You'll also need to add `$PREFIX/$HOST/usr/lib` for libraries like `libm.so`

Reply:
Can you also do this only for Linux?

Reply:
> Use $HOST instead of x86_64-conda_cos6-linux-gnu

Done.

> Can you add the directory $PREFIX/$HOST/usr/include as additionally to /usr/local/include?

This can be done but I'm not sure why, do you have anything in mind? I'm hesitant to keep `/usr/local/include` in the default search path to avoid conflicts.

> You'll also need to add $PREFIX/$HOST/usr/lib for libraries like libm.so

Sorry I don't understand what you mean?

Reply:
> This can be done but I'm not sure why, do you have anything in mind? I'm hesitant to keep /usr/local/include in the default search path to avoid conflicts.

I don't like changing the default behaviour of a package without a good reason.

> Sorry I don't understand what you mean?

You are adding $PREFIX/$HOST/usr/include because the compiler can't find the headers. What I'm saying is that the compiler can't find the libraries as well.

Reply:
I expect trying to mix libraries from conda with libries from `/lib` and `/usr/lib` will cause issues and it's better for the libraries and headers to not be found at all. If someone wants to they can still expliciy include/link with packages installed on the host.

Reply:
That said, I'm eager to get this merged so if still you disagree I'll add `/lib`, `/usr/lib` and `/usr/local/include` back it.

Reply:
I lean towards @chrisburr 's approach of trying to prevent mixing with `/lib` and `/usr/lib` as much as possible.

Reply:
> I expect trying to mix libraries from conda with libries from /lib and /usr/lib will cause issues and it's better for the libraries and headers to not be found at all.

When using the new gcc7 packages on a system with gcc7 as the compiler, they work fine. So, a compromise would be to not use `/usr/local/include` when `CONDA_BUILD` env variable is set. This is in line with other packages (like setuptools) where we disable behaviour that is harmful in a conda build environment. (We can also do this later and keep `/lib`, `/usr/lib`, etc if you are in a hurry)

Reply:
> When using the new gcc7 packages on a system with gcc7 as the compiler, they work fine.

Agreed, but none of the systems (SLC6, CentOS 7, Arch, Ubuntu) I have access to use gcc7 for the system compiler and I suspect this is the same for most people. Even if gcc7 was more common, I encourage people to use conda so they can move between different machines more easily and not be stuck the one machine where everything works for an unknown reason.

I think this is analgous to if conda's build of `python` should try and import packages from `/usr/lib/pythonX.Y/site-packages`.